api-participant/src/main/java/bio/terra/pearl/api/participant/service/EnrollmentExtService.java

```diff
@@ -0,0 +1,125 @@
+    this.portalEnvironmentService = portalEnvironmentService;
+    this.registrationService = registrationService;
+    this.studyEnvironmentService = studyEnvironmentService;
+    this.enrolleeService = enrolleeService;
+    this.participantUserService = participantUserService;
+  }
+
+  /**
+   * Enroll a governed user into a study. If the governed user is new, they will be registered and
+   * enrolled. If the governed user already exists, they will simply be enrolled.
+   *
+   * @param operator Proxy user
+   * @param portalShortcode Portal shortcode
+   * @param envName Environment name
+   * @param studyShortcode Study shortcode
+   * @param preEnrollmentId Pre-enrollment response ID
+   * @param governedPpUserId Governed pp user ID, or null if they need a user registered
+   * @return New enrollee
+   */
+  @Transactional
+  public HubResponse enrollGovernedUser(
+      ParticipantUser operator,
+      String portalShortcode,
+      EnvironmentName envName,
+      String studyShortcode,
+      UUID preEnrollmentId,
+      UUID governedPpUserId // could be null if a totally new user
+      ) {
+
+    PortalParticipantUser portalParticipantUser =
+        authUtilService
+            .authParticipantToPortal(operator.getId(), portalShortcode, envName)
+            .ppUser();
+
+    Enrollee proxy =
+        fetchOrCreateProxyEnrollee(operator, portalParticipantUser, studyShortcode, envName);
+
+    if (governedPpUserId == null) {
+      String governedUserName =
+          registrationService.getGovernedUsername(
+              operator.getUsername(), operator.getEnvironmentName());
+
+      return enrollmentService.registerAndEnrollGovernedUser(
+          envName,
+          studyShortcode,
+          proxy,
+          operator,
+          portalParticipantUser,
+          preEnrollmentId,
+          governedUserName);
+    } else {
+      PortalParticipantUser governedPpUser =
+          authUtilService.authParticipantUserToPortalParticipantUser(
+              operator.getId(), governedPpUserId);
+
+      ParticipantUser governedUser =
+          participantUserService
+              .find(governedPpUser.getParticipantUserId())
+              .orElseThrow(IllegalStateException::new);
+
+      return enrollmentService.enrollGovernedUser(
+          envName,
+          studyShortcode,
+          proxy,
+          operator,
+          portalParticipantUser,
+          governedUser,
+          governedPpUser,
+          preEnrollmentId);
+    }
+  }
+
+  private Enrollee fetchOrCreateProxyEnrollee(
+      ParticipantUser user,
+      PortalParticipantUser ppUser,
+      String studyShortcode,
+      EnvironmentName envName) {
+    return enrolleeService
+        .findByParticipantUserIdAndStudyEnv(ppUser.getParticipantUserId(), studyShortcode, envName)
+        .orElseGet(
+            () ->
+                enrollmentService
+                    .enroll(ppUser, envName, studyShortcode, user, ppUser, null, false)
```

Comment:
for my own understanding: we only attach the pre-enrollment response if the enrollee is the subject?

Reply:
Yes, this is accurate. That code/idea existed before I started working on proxies but I think the idea is that the proxy would only have a pre-enroll response if they enrolled themselves. That way, if the pre-enroll answers differed, or we needed to grab more information about the subject than just the name, then we could store those answers separately.